src/sink/file.cpp

```diff
@@ -120,20 +124,42 @@ auto file_t::filename(const record_t&) const -> std::string {
     return {data.path};
 }
 
+auto file_t::exists(const std::string& filename) const -> bool {
+    struct stat buf = {};
+    return ::stat(filename.c_str(), &buf) == 0;
+}
+
 auto file_t::backend(const std::string& filename) -> file::backend_t& {
     const auto it = data.backends.find(filename);
 
     if (it == data.backends.end()) {
-        auto stream = stream_factory->create(filename, std::ios_base::app);
-        auto flusher = flusher_factory->create();
+        return create_backend(filename);
+    }
 
-        return data.backends.insert(it,
-            std::make_pair(filename, file::backend_t(std::move(stream), std::move(flusher))))->second;
+    if (should_stat && !exists(filename)) {
```

Comment:
It checks only file existence at the original path, but doesn't check that backend's file was moved. So if someone moves the file and creates a new one at original path, `file_t` will not recreate backend.